tests/kernel/interrupt/src/nested_irq.c

```diff
@@ -5,6 +5,21 @@ struct k_timer timer;
 
 #define ISR0_OFFSET 1
 #define ISR1_OFFSET 2
+
+/* Keeping isr0 to be lowest priority than system timer
+ * so that it can be interrupted by timer triggered.
+ * In NRF5, RTC system timer is of priority 1 and
+ * in all other architectures, system timer is considered
+ * to be in priority 0.
+ */
+#if defined(CONFIG_ARM)
+	#define ISR0_PRIO 2
+	#define ISR1_PRIO 1
+#else
+	#define ISR0_PRIO 1
+	#define ISR1_PRIO 0
```

Comment:
The limit in ARC is 16 priority levels, if I am not mistaken.
And 0 is taken up by the timer. 
The expectation here is to have lower priority than timer.

Reply:
Is the timer defaulting to the **lowest** priority on ARC?  That seems wrong, designwise.  Usually you want timekeeping to happen at high priorities, with a few available above it for devices with latency requirements.

Also x86_64 (to be merged any moment) uses "raw" APIC priority values, meaning that a user interrupt needs to have a priority in the range 2-15 (the first two "priority classes" are are for CPU exceptions).  x86 does the adjustment for you and allows the use of priorities 0-13.

Reply:
Also also, xtensa has fixed priorities for each interrupt and ignores the priority argument entirely.  It won't work with this test unless you also select specific IRQs (though it looks like it doesn't run this subtest anyway, as there's no trigger_irq() implementation for it).